app/models/api/openid_connect/o_auth_application.rb

```diff
@@ -0,0 +1,66 @@
+module Api
+  module OpenidConnect
+    class OAuthApplication < ActiveRecord::Base
+      has_many :authorizations, dependent: :destroy
+      has_many :user, through: :authorizations
+
+      validates :client_id, presence: true, uniqueness: true
+      validates :client_secret, presence: true
+      validates :client_name, presence: true
+      validates_uniqueness_of :client_name, scope: :redirect_uris
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.